presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -254,6 +255,10 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                 new TransformCorrelatedSingleRowSubqueryToProject(),
                 new CheckSubqueryNodesAreRewritten(),
                 new PredicatePushDown(metadata, sqlParser),
+                new IterativeOptimizer(
+                        stats,
+                        ImmutableList.of(new PickLayout(metadata)),
```

Comment:
That means it will overwrite the layout chosen by add exchanges, which I assume is why the condition was there to begin with.  If it will always choose the same one anyway, then this is a good approach, but check on that first.

Reply:
Okay, I understand now. Previously `PickLayout / PickTableLayout` was used in tests _only_ (`PlanOptimizers`'s  `forceSingleNode` is tests' utility).

Now we change the plan: let `PickTableLayout` do its job always, but then it needs to run before Exchanges.
This is more than "extend" in the commit message hints at.

Change commit message to:

```
Execute PickTableLayout before AddExchanges

Previously, the `PickTableLayout` rule was responsible for picking layout in local test execution only.
However, in order to have better stats estimates, we need to know the table layouts earlier during the
planning. We change PickTableLayout so that it can pick (and re-pick) table layout iteratively.
Even though the rule follows `AddExchanges`'s logic, the intention is not to overwrite layout chosen
by `AddExchanges`. Therefore now the rule should not be run after `AddExchanges`.
```

Also, change `// must be run before AddExchanges` to `// must not run after AddExchanges`.